src/discussions/common/EmptyPage.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import classNames from 'classnames';
+import { useDispatch, useSelector } from 'react-redux';
+import { useRouteMatch } from 'react-router';
+
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button } from '@edx/paragon';
+
+import emptyIcon from '../../assets/empty.png';
+import { ALL_ROUTES, Routes } from '../../data/constants';
+import { useTotalThreadCount } from '../data/hooks';
+import { selectAreThreadsFiltered } from '../data/selectors';
+import messages from '../messages';
+import { messages as postMessages, showPostEditor } from '../posts';
+
+const TOPIC_PATH = Routes.TOPICS.PATH[0];
+const CATEGORY_PATH = Routes.TOPICS.PATH[1];
+const ROUTE_MESSAGES = {
+  [Routes.POSTS.MY_POSTS]: messages.emptyMyPosts,
+  [Routes.POSTS.ALL_POSTS]: messages.emptyAllPosts,
+  [TOPIC_PATH]: messages.emptyAllTopics,
+  [CATEGORY_PATH]: messages.emptyAllTopics,
+};
+
+const ROUTE_TITLES = {
+  [Routes.POSTS.MY_POSTS]: messages.noPostSelected,
+  [Routes.POSTS.ALL_POSTS]: messages.noPostSelected,
+  [TOPIC_PATH]: messages.noTopicSelected,
+  [CATEGORY_PATH]: messages.noTopicSelected,
+};
+
+function EmptyPage({ intl }) {
+  const match = useRouteMatch(ALL_ROUTES);
+  const dispatch = useDispatch();
+
+  const isFiltered = useSelector(selectAreThreadsFiltered);
+  const totalThreads = useTotalThreadCount();
+  let { emptyTitle } = messages;
+  let emptyMessage = ROUTE_MESSAGES[match.path] || '';
+  let displayMessage = true;
+  const isOnTopicsPage = Routes.TOPICS.PATH.includes(match.path);
+
+  function addPost() {
+    return dispatch(showPostEditor());
+  }
+
+  // If user is on topic page
+  // Show "no topic selected" if no topic selected yet
+  // Show "nothing here yet" if topic with no posts selected
+  // Show "no post selected" if viewing a topic with posts
+  if (isOnTopicsPage && match.params.topicId) {
+    emptyMessage = messages.emptyTopic;
+    emptyTitle = messages.emptyTitle;
+    if (isFiltered || totalThreads > 0) {
+      displayMessage = false;
+      emptyTitle = messages.noPostSelected;
+    }
+  } else if (isOnTopicsPage) {
+    displayMessage = false;
+    emptyTitle = ROUTE_TITLES[match.path];
+  } else if (isFiltered || totalThreads > 0) {
+    displayMessage = false;
+    emptyTitle = ROUTE_TITLES[match.path];
+  }
```

Comment:
This seems a bit hacky. I think a better approach would be to not centralise this logic. Have each part of the app show the emptypage component with its own specialised logic and just pass in the empty page message.

Reply:
True. I'll update.

Reply:
Looks like there isn't a way to completely get rid of this, but it does look cleaner.